firmware/user/modes/mode_color_movement.c

```diff
@@ -205,7 +205,8 @@ uint8_t cmButtonState = 0;
 uint8_t cmLastButtonState = 0;
 
 uint8_t cmBrightnessIdx = 2;
-uint8_t ledOrderInd[] = {LED_UPPER_LEFT, LED_LOWER_LEFT, LED_LOWER_MID, LED_LOWER_RIGHT, LED_UPPER_RIGHT, LED_UPPER_MID};
+uint8_t ledOrderInd[NUM_LIN_LEDS] = {LED_UPPER_LEFT, LED_LOWER_LEFT, LED_LOWER_MID, LED_LOWER_RIGHT, LED_UPPER_RIGHT, LED_UPPER_MID};
```

Comment:
should be ledOrderInd[6]
What is crucial is that the initialised values are from 0 to NUM_LIN_LEDS - 1

Reply:
You address it up to `NUM_LIN_LED` so it needs to be at least that long, otherwise it'll address out of bounds
```
    for (uint8_t i = 0; i < NUM_LIN_LEDS; i++)
    {
        //use axis colors or hue colors computed above
        leds[ledOrderInd[i]].r = ledr;
        leds[ledOrderInd[i]].g = ledg;
        leds[ledOrderInd[i]].b = ledb;
        //clear if going to cycle lights
        leds[ledOrderInd[i]].r  = 0;
        leds[ledOrderInd[i]].g  = 0;
        leds[ledOrderInd[i]].b  = 0;
    }
```

Reply:
I finally see what was worrying you. The offending code was a mistake. It is no longer in mode_color_movement.c. I’ll use ledOrderInd similar to line 524 in roll3.c when I integrate everything back into color movement.